pkg/engine/mutation.go

```diff
@@ -83,8 +92,47 @@ func Mutate(policyContext PolicyContext) (resp response.EngineResponse) {
 			resp.PolicyResponse.Rules = append(resp.PolicyResponse.Rules, ruleResponse)
 			incrementAppliedRuleCount()
 		}
+
+		// insert annotation to podtemplate if resource is pod controller
+		// skip inserting on existing resource
+		if reflect.DeepEqual(policyContext.AdmissionInfo, RequestInfo{}) {
+			continue
+		}
+
+		if strings.Contains(PodControllers, resource.GetKind()) {
+			var ruleResponse response.RuleResponse
+			ruleResponse, patchedResource = processOverlay(ctx, podTemplateRule, patchedResource)
```

Comment:
Would the annotation be also added for existing resources? 
When policy-contoller is processing values in the background. For mutation, we don't apply patches to the resource. 
With this, we will create a policy violation if the annotation is not present.

Shouldn't we add annotations in admission control only and not the policy-controller ?

Reply:
See: https://github.com/nirmata/kyverno/blob/2d022d457a48363c58ea224a681e92dfa6b83af0/pkg/engine/mutation.go#L97-L98